pages/__admin/messages/index.tsx

```diff
@@ -0,0 +1,81 @@
+import AdminMessageItem from "feature/admin-messages-list/AdminMessageItem"
+import ChannelMessageFunction from "databases/functions/ChannelMessages"
+import getConfig from "next/config"
+import init from 'nextjs-basic-auth'
+import useToastHook from "@hooks/toast/useToastHook"
+import { AdminChannelMessage } from "types/Feature/AdminMessagesList/AdminMessageItem"
+import { GetServerSidePropsContext } from "next"
+import { getAllAdminMessages } from "@services/message/adminMessageService"
+import { useRouter } from "next/router"
+import { useState } from "react"
+
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
+    const {serverRuntimeConfig} = getConfig()
+    const {query} = context
+
+    const users = [
+        {user: serverRuntimeConfig.BETTER_BACKDOOR_ADMIN_USER, password: serverRuntimeConfig.BETTER_BACKDOOR_PASSWORD},
+    ]
+
+    const basicAuthCheck = init({
+        users
+    })
+
+    await basicAuthCheck(context.req, context.res)
+
+    const offset = query?.offset ? parseInt(query?.offset as string) : 0
+    const channelMessages = await ChannelMessageFunction.adminGetChannelMessage(10, offset)
+    
+    return {
+        props: {
+            messages: channelMessages
+        }
+    }
+}
+
+export type AdminMessageScreenProps = {
+    messages: AdminChannelMessage[]
+}
+
+const AdminMessagesScreen = (props: AdminMessageScreenProps) => {
+    const [data, setData] = useState<AdminChannelMessage[]>(props?.messages)
+    const [isLoadingMessages, setIsLoadingMessages] = useState(false)
+    const router = useRouter()
+    const offset = router?.query?.offset as string
+
+    const {noNewMessagesToast} = useToastHook()
+
+    const loadMoreMessages = async() => {
+        const response = await getAllAdminMessages(10, parseInt(offset || '10'))
+        if(response?.data?.length === 0) return noNewMessagesToast()
+
+        setData(data => [...data, ...response?.data])
+        router.query.offset = response?.next?.toString()
+        return router.replace({
+            query: {
+                ...router.query,
+                offset: response?.next?.toString()
+            }
+        })
+    }
+
+    const onLoadMoreClicked = async() => {
+        setIsLoadingMessages(true)
+        await loadMoreMessages()
+        setIsLoadingMessages(false)
+    }
+
+    return <div className="p-4">
+        <p className="text-xl font-inter">Confessions List</p>
+        <div className="flex flex-row flex-wrap w-full gap-3 mt-8">
+        {data?.map((item) => {
+            return <AdminMessageItem item={item} key={item?.message_id}/>
+        })}
+        </div>
+        <button className="w-full bg-primaryBlue text-white px-4 py-4 rounded-md mt-8" onClick={() => onLoadMoreClicked()}>
+            {isLoadingMessages ? 'Loading...' : 'Load More Messages'}
+        </button>
+    </div>
+}
+
+export default AdminMessagesScreen
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `AdminMessagesScreen` component, the `loadMoreMessages` function updates the router's query parameters directly. This could potentially cause issues with React's state updates. Instead, consider using the `useRouter` hook's `push` or `replace` methods to update the query parameters.

```diff
-        router.query.offset = response?.next?.toString()
-        return router.replace({
-            query: {
-                ...router.query,
-                offset: response?.next?.toString()
-            }
-        })
+        const newOffset = response?.next?.toString();
+        return router.replace({
+            pathname: router.pathname,
+            query: { ...router.query, offset: newOffset },
+        });
```

Also, the `loadMoreMessages` function does not handle any errors that might occur during the fetching of messages. Consider adding error handling to improve the robustness of your code.

```diff
-        const response = await getAllAdminMessages(10, parseInt(offset || '10'))
-        if(response?.data?.length === 0) return noNewMessagesToast()
+        try {
+            const response = await getAllAdminMessages(10, parseInt(offset || '10'))
+            if(response?.data?.length === 0) return noNewMessagesToast()
+        } catch (error) {
+            // Handle or display the error appropriately
+        }
```

The `onLoadMoreClicked` function sets `isLoadingMessages` to `true`, then calls `loadMoreMessages`, and finally sets `isLoadingMessages` to `false`. If `loadMoreMessages` throws an error, `isLoadingMessages` will remain `true`. To ensure `isLoadingMessages` is set to `false` even when an error occurs, consider using a `finally` block.

```diff
-        setIsLoadingMessages(true)
-        await loadMoreMessages()
-        setIsLoadingMessages(false)
+        setIsLoadingMessages(true);
+        try {
+            await loadMoreMessages();
+        } finally {
+            setIsLoadingMessages(false);
+        }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->